src/V3Force.h

```diff
@@ -12,17 +12,12 @@
 // Version 2.0.
 // SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0
 //
-
 //*************************************************************************
 
 #ifndef VERILATOR_V3FORCE_H_
 #define VERILATOR_V3FORCE_H_
 
-#include "config_build.h"
-#include "verilatedos.h"
-
-#include "V3Error.h"
-#include "V3Ast.h"
```

Comment:
I'm a bit struggling to see the point of adding unused include files. Also static analysis tools flag them as unused.

Reply:
Added these back though

Reply:
The first two I want because if I need some portability thing to be added I want it to apply everywhere, not have to edit to add a new include.

The other two are ok removing - but similar comment I'll make below - do it everywhere as separate patch first.

BTW what tool complains about them as extra?  FWIW I run static on only the .cpp files and perhaps should do better - maybe we should add a github action to do more formal.

Reply:
Also BTW some visitors don't take AstNetlist* nodep as an argument and use the top one directly.  I think that's probably cleaner, so you can then remove "class AstNetlist;" also.  For example see V3CUse.

If you prefer I clean up the extra includes let me know.

Reply:
I mean I'm suggesting

    static void apply() { ForceConvertVisitor{v3Global.rootp()}; }

Reply:
This also makes it much clearer when a visitor is unusual in that it operates on other than a netlist.

Reply:
> BTW what tool complains about them as extra?

CLion inspections flag unused headers

> FWIW I run static on only the .cpp files and perhaps should do better

In my [entirely subjective and arguably strange] world, Header files should be sufficient (i.e.: does not depend on an earlier include at use site), and necessary (only contain stuff required for the declarations in the header file). Everything else hinders understanding of the code. You can check "sufficiency" by compiling the .h file as if it was a .c file (also, this should produce an empty object file, otherwise you have duplicate static code/data instanes). The 'necessary' can be checked by static analysis (or whatever compiler option?). 

Fair enough about the portability thing, though I would still prefer to sign those off when applied rather than blast it everywhere by default, but I don't feel strongly about this.

> maybe we should add a github action to do more formal

Probably a good idea.

> Also BTW some visitors don't take AstNetlist* nodep as an argument and use the top one directly. I think that's probably cleaner, so you can then remove "class AstNetlist;" also. For example see V3CUse.
> I mean I'm suggesting
> static void apply() { ForceConvertVisitor{v3Global.rootp()}; }
> This also makes it much clearer when a visitor is unusual in that it operates on other than a netlist.

Please don't. I think this is a step in the wrong direction (and I have secretly been working on un-doing it since I started contributing, busted!). Long term we will likely want to use parallel algorithms to speed up Verilator internals. Global state is the number one pain point in doing so. It also hinders other more interesting algorithms (e.g.: this is why hierarchical mode needs to fork, which is then a pain to communicate through). I appreciate many many things rely on global state now, but we should not dig ourselves even further in.

Reply:
I agree to having headers clean, and isolatable, beyond wanting the verilatedos included in every file for reason I mentioned. I haven't tried CLion, I will.

I'm not sure what model you are thinking for parallelism and how much having netlist as an argument would realistically help, but will accept it hides information and you think it's a step back.  BTW the original reason for that was to handle multiple netlists being present simultaneously similar to Verilog-Perl, which never was needed (or not so far).

Reply:
I think CLion is just the IDE and it runs cpplint underneath.

I committed some header cleanups to avoid those includes.

Reply:
Correct, it runs a bunch of stuff though, cpplint is probably one of them, there is also clang-tidy